src/nagini_translation/lib/program_nodes.py

```diff
@@ -565,6 +565,24 @@ def get_func_or_method(self, name: str) -> Optional['PythonMethod']:
             return self.get_function(name)
         else:
             return self.get_method(name)
+        
+    def get_compatible_func_or_method(self, name: str, arg_types: List[PythonType]) -> Optional['PythonMethod']:
+        func = self.get_func_or_method(name)
+        if not func:
+            return None
+        
+        param_types = [arg.type for arg in func.get_args()]
+        if len(param_types) != len(arg_types):
+            return None
```

Comment:
Is this case ever supposed to happen or would it indicate an internal error?

Reply:
Thinking about this again, I'm not sure if this could ever even happen at all.
Checking for matching argument types is still necessary though.